### PR TITLE
feature/Request for getting upgrade options for a BM

### PR DESCRIPTION
**Pull request by mfbmina**
Added support for feature that gets upgrade options for a BM.

I didn't write any usage examples because it already exists at examples.

On the tests, I created all the necessary tests and correct others who are broken or not working properly.

### CODE REVIEW

**Comment by mfbmina** [comment]
@fernandes or @cphrmky , is there any problem here?

File: lib/fog/softlayer/requests/compute/get_virtual_guest_upgrade_item_prices.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Mock
         # @return [Excon::Response]
         def get_virtual_guest_upgrade_item_prices(id)
           response = Excon::Response.new
-          found = self.get_vms.body.map{|server| server['id']}.include?(id)
+          found = self.get_vms.body.map{|server| server['id']}.include?(id.to_s)
```
Comment (fernandes) [inline lib/fog/softlayer/requests/compute/get_virtual_guest_upgrade_item_prices.rb]: why added `#.to_s` here?

Comment (mfbmina) [inline lib/fog/softlayer/requests/compute/get_virtual_guest_upgrade_item_prices.rb]: in test of requests, id was a string and in server's it was an integer. To work on both, i'm always change it to string.

**Comment by cphrmky** [comment]
Looks good by me!  :+1:

File: lib/fog/softlayer/requests/compute/get_virtual_guest_upgrade_item_prices.rb (hunk 1 above)
Comment (fernandes) [inline lib/fog/softlayer/requests/compute/get_virtual_guest_upgrade_item_prices.rb]: great! lets merge! :+1:

**Pull request merged**